### PR TITLE
Added `decorator` to `DatepickerInput` component

### LINKED ISSUE #17991: Decorator: DatePicker

### PR DESCRIPTION
**Pull request by guidari**
Closes #17991

Added decorator to `Datepicker` 

#### Changelog

**New**

- New `decorator` prop added to `DatepickerInput` component

**Changed**

- Deprecated `slug` prop
- Update rendering slug to render the decorator component but still set the size if it's an AILabel
- Updated test file
- Added an `block-size` to the style to fix the alignment _(this fix can be used on components that are inside an input)_

#### Testing / Reviewing

- Make sure the `Datepicker` still work as expected and the one with `decorator` also works fine

- [ ] Delete `DatepickerInput` with `Tooltip`

### CODE REVIEW

File: packages/styles/scss/components/date-picker/_date-picker.scss [hunk 1]
```diff
@@ -191,14 +191,28 @@
   }
 
   // Styles for `AILabel` rendered inside `DatePickerInput`
+  .#{$prefix}--date-picker-input__wrapper--decorator
+    .#{$prefix}--date-picker-input-inner-wrapper--decorator,
```
Comment (ariellalgilmore) [inline packages/styles/scss/components/date-picker/_date-picker.scss]:
```suggestion
    .#{$prefix}--date-picker-input-inner-wrapper--decorator > *,
```

File: packages/styles/scss/components/date-picker/_date-picker.scss [hunk 2]
```diff
@@ -191,14 +191,29 @@
   }
 
   // Styles for `AILabel` rendered inside `DatePickerInput`
+  .#{$prefix}--date-picker-input__wrapper--decorator
+    .#{$prefix}--date-picker-input-inner-wrapper--decorator
+    > *,
+  .#{$prefix}--date-picker-input__wrapper--decorator .#{$prefix}--decorator,
```
Comment (ariellalgilmore) [inline packages/styles/scss/components/date-picker/_date-picker.scss]:
```suggestion
```
the wrapper above should handle everything now

**Review by ariellalgilmore — APPROVED** [review]
Awesome! LGTM!

**Review by alisonjoseph — APPROVED** [review]
nice!

File: packages/react/src/components/DatePickerInput/DatePickerInput.tsx [hunk 3]
```diff
@@ -38,6 +39,11 @@ export interface DatePickerInputProps
    */
   datePickerType?: 'simple' | 'single' | 'range';
 
+  /**
+   * **Experimental**: Provide a `decorator` component to be rendered inside the `RadioButton` component
```
Comment (alisonjoseph) [inline packages/react/src/components/DatePickerInput/DatePickerInput.tsx]:
```suggestion
   * **Experimental**: Provide a `decorator` component to be rendered inside the `DatePickerInput` component
```

Comment (guidari) [reply packages/react/src/components/DatePickerInput/DatePickerInput.tsx]: Ohh! Copy and paste 🤦‍♂️
Thanks haha

**Review by riddhybansal — COMMENTED** [review]
This LGTM

**Review by riddhybansal — APPROVED** [review]

**Pull request merged**